test/unit/metadata_test.py

```diff
@@ -194,16 +195,84 @@ def test_frictionless_data_package_non_empty():
     assert len(datapackage.resources) == len(RESOURCE_METADATA)
 
 
+METADATA_OVERRIDE_KEYS = [
+    "layer_code",
+    "table_type_code",
+    "timeseries_resolution_code",
+    "additional_summary_text",
+    "additional_layer_text",
+    "additional_source_text",
+    "additional_primary_key_text",
+    "additional_details_text",
+]
+
+
 def test_frictionless_data_package_resources_populated():
     datapackage = PUDL_PACKAGE.to_frictionless()
     for resource in datapackage.resources:
         assert resource.name in RESOURCE_METADATA
         expected_resource = RESOURCE_METADATA[resource.name]
-        assert expected_resource["description"] == resource.description
+        # TODO: remove str option after metadata migration
+        strings_to_find = []
+        if isinstance(expected_resource["description"], str):
+            strings_to_find.append(expected_resource["description"])
+        else:
+            for k in METADATA_OVERRIDE_KEYS:
+                if k in expected_resource["description"]:
+                    strings_to_find.append(expected_resource["description"][k])
+        assert any(
+            resource.description.find(candidate) >= 0 for candidate in strings_to_find
+        )
         assert expected_resource["schema"]["fields"] == [
             f.name for f in resource.schema.fields
         ]
         assert (
             expected_resource["schema"].get("primary_key", [])
             == resource.schema.primary_key
         )
+
+
+description_compliant_tables = [
```

Comment:
discussed offline: let's key off of "is `resource_dict["description"]` a dict?" and not have to manually maintain this list.